charts/consul/templates/connect-inject-clusterrole.yaml

```diff
@@ -33,6 +33,17 @@ rules:
   - get
   - list
   - update
+{{- if (and .Values.global.secretsBackend.vault.enabled .Values.global.secretsBackend.vault.consulConnectInjectCARole .Values.global.secretsBackend.vault.connectInject.tlsCert.secretName  .Values.global.secretsBackend.vault.connectInject.caCert.secretName)}}
+- apiGroups:
```

Comment:
k8s permissions needed when web-cert-manager is turned off and connect injector manages webhook certs itself with vault.